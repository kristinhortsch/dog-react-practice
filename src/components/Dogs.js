import React from 'react';
import Dog from './Dog';
import PropTypes from 'prop-types';

function Dogs({ dogs }) {
  const listOfDogs = dogs.map(dog => {
    return <li key={dog.name}><Dog dog={dog}/></li>;
  });

  return (
    <div>
      <h3>Dog List</h3>
      <ul>
        {listOfDogs}
      </ul>
    </div>
  );
}

Dogs.propTypes = {
  dogs: PropTypes.array.isRequired
};

export default Dogs;
