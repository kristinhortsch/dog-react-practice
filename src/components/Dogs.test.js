import React from 'React';
import renderer from 'react-test-renderer';
import Dogs from './Dogs';

describe('Characters', () => {
  it('matches a snapshot', () => {
    const dogs = ['rover']; 
    const tree = renderer.create(<Dogs dogs={dogs}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
