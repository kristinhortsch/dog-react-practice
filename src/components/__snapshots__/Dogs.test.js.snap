// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshot 1`] = `
<div>
  <h3>
    Dog List
  </h3>
  <ul>
    <li>
      <h2 />
      <p>
        Age: 
      </p>
      <p>
        Weight: 
         lbs
      </p>
    </li>
  </ul>
</div>
`;
